app/lib/measurements/creator.rb

```diff
@@ -0,0 +1,26 @@
+module Measurements
+  class Creator
+    def initialize(payload)
+      @captured_at = payload.delete(:captured_at).to_datetime
+      @device_id = payload.delete(:device_id)
+      @location = generate_location(
+        longitude: payload.delete(:longitude),
```

Comment:
Can `extract!` be used here like `generate_location(payload.extract!(:longitude, :latitude))` ?